tests/sys/shell_scripting/main.c

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2024 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup tests
+ * @{
+ *
+ * @file
+ * @brief   Shell Scripting Test Application
+ *
+ * @author  Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ * @}
+ */
+
+#include <stdlib.h>
+
+#include "shell.h"
+#include "vfs_default.h"
+#include "vfs_util.h"
+#include "ztimer.h"
+
+#ifndef SCRIPT_FILE
+#define SCRIPT_FILE VFS_DEFAULT_DATA "/script.sh"
+#endif
+
+static int cmd_msleep(int argc, char **argv)
+{
+    if (argc != 2) {
+        return -1;
+    }
+
+    ztimer_sleep(ZTIMER_MSEC, atoi(argv[1]));
+    return 0;
+}
+
+static int cmd_echo(int argc, char **argv)
+{
+    for (int i = 1; i < argc; ++i) {
+        printf("%s ", argv[i]);
+    }
+    puts("");
+
+    return 0;
+}
+
+static int cmd_add(int argc, char **argv)
+{
+    int sum = 0;
+
+    for (int i = 1; i < argc; ++i) {
+        sum += atoi(argv[i]);
+    }
+
+    printf("%d", sum);
+    return 0;
+}
+
+static const shell_command_t shell_commands[] = {
+    { "msleep", "sleep for a number of ms", cmd_msleep },
+    { "echo", "echo parameters to console", cmd_echo },
+    { "add", "add up a list of numbers", cmd_add },
+    { NULL, NULL, NULL }
+};
+
+static int _create_script(void)
+{
+    const char file[] = {
```

Comment:
Add a `"# comment"` and maybe also a `"\n"` to test that nothing unexpected happens in `readline`.

Reply:
And you could also add a test for the case that a command fails.

Reply:
and a test to check what happens when the line is too long